transport_plugins/jlink/iotile_transport_jlink/jlink_background.py

```diff
@@ -222,6 +222,38 @@ def _read_ff_dump_resp(self, length):
         flash_dump = self._read_memory(buffer_addr, length)
         return flash_dump
 
+    def _read_mapped_memory(self, device_info, region, read_start_addr, read_length):
+        if read_length is None:
+            if region == 'ram':
+                memory = self._read_memory(device_info.ram_start, device_info.ram_size)
+            elif region == 'flash':
+                memory = self._read_memory(device_info.flash_start, device_info.flash_size)
+            elif region == 'mapped':
+                raise ArgumentError("Must specify a data length to read from mapped memory.")
+        else:
+            if region == 'ram':
+                if read_length > device_info.ram_size:
+                    raise ArgumentError("Invalid data length to read from RAM.")
+                if read_start_addr > device_info.ram_size:
+                    raise ArgumentError("Invalid start address to read from RAM.")
+                memory = self._read_memory(read_start_addr + device_info.ram_start, read_length)
+
+            elif region == 'flash':
+                if read_length > device_info.flash_size:
+                    raise ArgumentError("Invalid data length to read from flash.")
+                if read_start_addr > device_info.flash_size:
+                    raise ArgumentError("Invalid start address to read from flash.")
+                memory = self._read_memory(read_start_addr + device_info.flash_start, read_length)
+
+            elif region == 'mapped':
+                if read_length > (device_info.ram_start + device_info.ram_size):
```

Comment:
These checks are too restrictive.  The mapped address space of a Cortex M4 is 4GB large so I would just verify that `read_start_addr` is > 0 and let the user go to town.

Reply:
Fixed.